src/DataStructures/Tags.hpp

```diff
@@ -0,0 +1,29 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/Metafunctions.hpp"
+
+/// \cond
+class DataVector;
+class ModalVector;
+/// \endcond
+
+namespace Tags {
+/// Given the `NodalTag` swap the `Tensor::type` with a `ModalVector`.
+template <typename NodalTag>
+struct Modal : db::SimpleTag {
+  static_assert(cpp17::is_same_v<typename NodalTag::type::type, DataVector>,
```

Comment:
Assert also that `db::item_type<NodalTag>` is a `Tensor`?

Reply:
The tensor type swap will fail to compile if that's not true, so I'd rather keep compile time down, personally.